testsuite/MDAnalysisTests/analysis/test_hbonds.py

```diff
@@ -162,6 +162,28 @@ def setUp(self):
         self.values['acceptor_resnm'] = np.array([], dtype="<U3")
 
 
+def build_parameters():
```

Comment:
Could you instead define `protein`, `nothing` and the `itertools.product` generator as class attributes of  `TestHydrogenBondAnalysisChecking`? This way the the `itertools.product` will be available when the decorator is used, and there won't be any need for the `build_parameters` functions.